### DISCUSSION TITLE
Downloading Prisma engines for Node-API for windows getaddrinfo ENOTFOUND 59996 Error

### DISCUSSION THREAD

**Discussion by akintolafemi** [prisma/prisma#18873] [Q&A] [answered]
Hi Everyone, 
I have been stuck on this problem for a couple of days now and solutions found online are not helpful. I have a nestjs project and installed prisma on. Running the command "npx prisma generate", I keep getting the error below

 Downloading Prisma engines for Node-API for windows [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a/windows/migration-engine.exe.gz.sha256 failed, reason: getaddrinfo ENOTFOUND 59996

From the searches I have seen, some said maybe my laptop is being a proxy server, but I am not. I have disabled proxy, switched to different networks but still getting same issue. I have tried with different versions of NodeJs but still the same. 

My current node version is 18.14.2 and npm 9.6.5.
I am using a Windows 11 OS.

I would appreciate some help here.

Thanks everyone

**Comment by nurul3101** [comment]
Hello @akintolafemi 👋 

If you click on this prisma binary link: https://binaries.prisma.sh/all_commits/1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a/windows/migration-engine.exe.gz.sha256 in your browser, does the engine get downloaded?

**Reply by akintolafemi** [reply]
Yes it does. @nurul3101

**Reply by akintolafemi** [reply]
I think it is a problem with windows operating system only. I moved the code to a Linux OS, ran same command "npx prisma generate" and it worked.

Back on windows, it doesn't work. I have checked everywhere for proxy settings but the problem does not go away. I am able to download the file from the url  [https://binaries.prisma.sh/all_commits/1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a/windows/migration-engine.exe.gz.sha256](url)

![image](https://user-images.githubusercontent.com/22343293/234592027-974fd60d-6920-4a22-92d6-e8584cb9eac8.png)

**Reply by nurul3101** [reply]
The error message suggests that there's a problem with the DNS resolution `(getaddrinfo ENOTFOUND)`.
Can you try to Flush your DNS cache? You can open a Command Prompt with administrator privileges and run the following command

`ipconfig /flushdns`

**Reply by akintolafemi** [reply]
Hi @nurul3101 I just did that now, still same error.
I have even switched between 3 different WiFis

**Reply by nurul3101** [reply]
Ah, okay! To confirm, you are using Prisma version 4.13.0, right?

**Reply by nurul3101** [reply]
Hey @akintolafemi 👋 
Did you had a chance to look at my last message? Did you found out the root cause?

**Reply by akintolafemi** [reply]
@nurul3101 Good day. Sorry for the late reply. Other projects got me.
Yes

"prisma": "^4.13.0"

**Reply by akintolafemi** [reply]
I did this ipconfig /flushdns but still no luck. Now, I'm guessing it's not just a prisma problem but between windows operating system and NodeJs. I switched to linux, moved the code base there and ran npx prisma generate and it worked perfectly as it should.

A friend complained about same issue last week but not with prisma. In his own case, it was expo. He also uses windows.

**Reply by nurul3101** [reply]
Thanks for getting back and providing context. It could be a Windows OS Issue. I would recommend also checking if this distributable is downloaded on your Windows machine: https://www.prisma.io/docs/reference/system-requirements#windows-runtime-dependencies
It's worth checking if downloading and executing the dependency fixes the issue or not.

**Reply by akintolafemi** [reply]
Alright @nurul3101 Will check it out and revert back here.

**Reply by akintolafemi** [reply]
@nurul3101 
Yes, I have it downloaded and installed. Please see screenshot below. Or do I need to remove the older ones?
Thanks.
![image](https://user-images.githubusercontent.com/22343293/236475501-dd3a873a-4775-4159-8127-727c5cf8345e.png)

**Reply by nurul3101** [reply]
You wouldn't need to remove the older ones. But installing the distributable also didn't fix the issue right?

**Reply by akintolafemi** [reply]
Hi @nurul3101 
The issue isn't fixed yet.

**Reply by janpio** [reply]
If you can download the file manually in your browser, but not when running in you console/terminal/cmd, then it means that is somehow configured to forbid these downloads. Especially that the error message mentions `59996`, which is not part of the actual URL indicates that something is configured to not directly try to download the file, but probably tries to use a proxy or similar (as you said in your initial message).

Can you try on another computer maybe?
Or with your existing computer via a different network/wifi?

**Reply by akintolafemi** [reply]
Hi @janpio Yes, I've tried different network/wifi. Still doesn't work. What I've done is to move the code to linux on same computer (using Virtual box) and it works fine.
Perhaps, I'll try another computer running on Windows.

Thank you for the help.

**Reply by nurul3101** [reply]
Let us know in case the issue gets fixed on another computer

**Reply by Secret1007** [reply]
> If you can download the file manually in your browser, but not when running in you console/terminal/cmd, then it means that is somehow configured to forbid these downloads. Especially that the error message mentions `59996`, which is not part of the actual URL indicates that something is configured to not directly try to download the file, but probably tries to use a proxy or similar (as you said in your initial message).
> 
> Can you try on another computer maybe? Or with your existing computer via a different network/wifi?

can i skip this download step, my computer is restricted with the internet connection

**Comment by HoseinGhanbari** [comment]
getting the same error using `"@prisma/client": "^4.15.0"` & `"prisma": "^4.15.0",` on `Windows 11`

```
D:\program>npx prisma generate
> Downloading Prisma engines for Node-API for windows [                    ] 0%
Error: request to https://binaries.prisma.sh/all_commits/8fbc245156db7124f997f4cecdd8d1219e360944/windows/migration-engine.exe.gz.sha256 failed, reason: getaddrinfo ENOTFOUND 10809
```

- set or unset `npm-proxy-settings` would not help.
- using external `tunnel / vpn / proxy` would not help.
- set or unset `NODE_TLS_REJECT_UNAUTHORIZED` would not help.
- changing `npm-registry-settings` would not help.

these are commands used so far:

```
npm config set proxy http://localhost:10809
npm config set https-proxy http://localhost:10809
npm config set strict-ssl false
npm config set registry http://registry.yarnpkg.com/
```

```
npm config delete proxy
npm config delete https-proxy
npm config delete strict-ssl
npm config delete registry
```

these are download links which are downloadable from browser but generates `getaddrinfo ENOTFOUND 10809` in the CMD.

```
https://binaries.prisma.sh/all_commits/8fbc245156db7124f997f4cecdd8d1219e360944/windows/migration-engine.exe.gz
https://binaries.prisma.sh/all_commits/8fbc245156db7124f997f4cecdd8d1219e360944/windows/migration-engine.exe.gz.sha256
```

it's also worth mentioning that using the same network and the same project but on `windows 10`, the `npx prisma generate` will run correctly without any error. so it seems the error some how related to `windows 11`.

any help would be appreciated.

**Reply by janpio** [reply]
I can unfortunately only answer the same as above: https://github.com/prisma/prisma/discussions/18873#discussioncomment-5927340 Replace port `59996` from that message with your `10809`.

**Reply by akintolafemi** [reply]
yes, @HoseinGhanbari this issue seems to be peculiar with Windows 11. Please let's know if you're able to sort it out. 
Thanks.

**Reply by HoseinGhanbari** [reply]
No working solution so far, I must switch to a win10 virtual machine whenever I need to update prisma related dependencies  😕

**Reply by akintolafemi** [reply]
@HoseinGhanbari sorry about your troubles. I'm not sure but you might want to ask a network engineer to help you check your machine provided you're stuck with windows 11. Switching between windows 11 and 10 on VM for sake of running Prisma dependencies doesn't look cool and can be tiring if your project is a large one. [https://github.com/prisma/prisma/discussions/18873#discussioncomment-6593289](url)

**Reply by janpio** [reply]
@HoseinGhanbari Are you getting exactly the same error message with `ENOUTFOUND 59996` error?

**Reply by HoseinGhanbari** [reply, marked as answer]
Thank you for your reply. The issue has been fixed on my recent windows and firewall update. I can now work with prisma related dependencies without leaving my current os. That was really annoying.

**Reply by OctoDino** [reply]
I just reinstalled prisma via npm, i got still errors, but after frustrated trying and "spaming" the command `npx prisma generate` it worked somehow... i can't explain why.
![grafik](https://github.com/prisma/prisma/assets/58665605/08665a58-a909-44b5-bb81-0595986449ff)

**Comment by akintolafemi** [comment]
Hi everyone. Sorry I am just coming back to this. Unfortunately, this issue is an operating system issue. Windows 11 to be precise. As I stated above, I tried every possible thing I found on the internet, did network reset, flush, changed npm proxy settings. None worked. I had the same getaddrinfo ENOTFOUND port problem with an expo (react-native) project. Although, I was certain the problem had nothing to do with NPM directly but some systems OS setting. I wasn't prepared to waste further time on it or maybe worst still, reinstall the windows. I just switched to my MAC and things are working as they should. 

I'd really love to see someone figure out why npm keeps complaining about getaddrinfo ENOTFOUND port on windows OS.
Good luck 🤞 

cc: @nurul3101

**Reply by janpio** [reply]
I am using Prisma on Windows 11, and many others are as well - so this is definitely not a universal problem. As mentioned before, _something_ on your Windows machine is routing traffic through something with port `59996` and then at that port nothing is forwarding that traffic to where it actually should go.

**Comment by shuang54** [comment]
I am currently facing the same problem, but I did not encounter it a month ago

**Reply by janpio** [reply]
Exactly the same error message with `ENOUTFOUND 59996` error?

**Reply by dianamoreno** [reply]
It was a network issue for me. I deactivated my firewall and change my network  to other one and it worked!

**Reply by shuang54** [reply]
What I got was this error, my version  
npm -v  9.6.7
node -v  v18.12.1
`
>npx prisma init
> Downloading Prisma engines for Node-API for windows [                    ] 0%Error: Invalid URL`

**Reply by janpio** [reply]
So you have a different problem (which is probably that an env var is configured wrongly).
Can you try Prisma 5.1 though? That should improve your error message or even prevent the problem.

**Reply by huaiguoguo** [reply]
i also facing same problem!

**Reply by shuang54** [reply]
My 'npx Prisma init' is running, and this is what I do:
Modified the 'Prisma': '^ 5.1.0', '@ Prisma/client': '^ 5.1.0' in package.json, and then ran NPM i. The version number before the modification was 5.0.0, and it worked normally after the modification.
I hope it will be helpful to you

**Comment by akintolafemi** [comment]
@nurul3101 
I thought to update this so this discussion could probably be closes.

Like I mentioned right, I was using windows 10 on same machine and prisma worked fine until I changed the OS to windows 11. From @HoseinGhanbari @dianamoreno it's a firewall problem on windows 11. 
So for anyone who runs into this issue and on windows 11. The first step is deactivate your firewall, restart your machine and retry whatever prisma cmd you wanted. Disabling firewall isn't a good practice anyways. And hopefully, a windows 11 update fixes the problem or folks at @prisma fixes it.

Thanks guys! Cheers!

**Comment by RANJEETJ06** [comment]
got error 

**Downloading Prisma engines for Node-API for windows [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/475c616176945d72f4330c92801f0c5e6398dc0f/windows/query_engine.dll.node.gz.sha256 failed, reason: unable to verify the first certificate**

**Reply by janpio** [reply]
Does this problem persist? If so, open a new issue or discussion please. Thanks.

**Reply by RANJEETJ06** [reply]
yep

**Reply by nvhuy5** [reply]
The same issue
C:\Users\nvhuy5>npx prisma -v
> Downloading Prisma engines for Node-API for windows [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5/windows/query_engine.dll.node.gz.sha256 failed, reason: unable to verify the first certificate

**Reply by nvhuy5** [reply]
I have solved this error, it is a network error

**Comment by Eyuelb** [comment]
I found a solution .Am on windows 11 just run ur ide with administrator the error will be gone

**Comment by 18601673727** [comment]
```
Downloading Prisma engines for Node-API for windows [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/5a9203d0590c951969e85a7d07215503f4672eb9/windows/query_engine.dll.node.sha256 failed, reason: read ECONNRESET
```
Still happening

**Comment by reinaldo-vombo** [comment]
It was a network issue for me, my internet was off, wen turn on  i run the generate and work fine

**Comment by Gugustinette** [comment]
I get a similar error in this environment :
- windows : `11 - 23H2 - 22631.5039`
- powershell : `5.1.22621.4391`
- node : `23.2.0`

Running `npx prisma init --datasource-provider sqlite --output ../generated/prisma` will result in :
```
> Downloading Prisma engines for Node-API for windows [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/f676762280b54cd07c770017ed3711ddde35f37a/windows/query_engine.dll.node.sha256 failed, reason: connect ETIMEDOUT 172.67.4.173:443
```

Note that the following PowerShell script works correctly :
```powershell
try {
  $Response = Invoke-WebRequest -Uri "https://binaries.prisma.sh/all_commits/f676762280b54cd07c770017ed3711ddde35f37a/windows/query_engine.dll.node.sha256" -UseBasicParsing -ErrorAction Stop
  Write-Host "Successfully connected to binaries.prisma.sh"
  Write-Host "Status Code: $($Response.StatusCode)"
  Write-Host "Content Length: $($Response.RawContentLength)"
  Write-Host "Content : $($Response)"
}
catch {
  Write-Host "Failed to connect to binaries.prisma.sh"
  Write-Host "Error: $($_.Exception.Message)"
}
```
Output :
```
Successfully connected to binaries.prisma.sh
Status Code: 200
Content Length: 98
Content : bf55a2a0ab04c961754894e040b72fea28feb2c3fb73e34fc2d04bdba4143a26  ./windows/query_engine.dll.node
```

The URL should be accessible from the CLI, but Prisma struggle with it for some reason 🤔

**Reply by Gugustinette** [reply]
So after a lot of digging inside Prisma's source code, I found that the error comes from the use of [`http-proxy-agent`](https://www.npmjs.com/package/http-proxy-agent) [here](https://github.com/prisma/prisma/blob/0752d425dc86190621b38f733899b93c78902bc2/packages/fetch-engine/src/getProxyAgent.ts#L105), which creates a `URL` from `HTTP_PROXY` or `HTTPS_PROXY` [here](https://github.com/TooTallNate/proxy-agents/blob/536aaa5165b888c548cbfa8095801c05478605fe/packages/http-proxy-agent/src/index.ts#L52).

In my case, the problem is I'm using my company's proxy manually like so :
```
$env:HTTPS_PROXY="http://[USERNAME]:[PASSWORD]@proxy.my-company.com:80"
```

I forgot to encode my password as a URL (using [`encodeURIComponent`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) for example), and as it contains a few `@`s, it broke the URL parsing.

**Comment by atsuunimog** [comment]
Update Prisma CLI and Client. This worked for me
npm install prisma@latest --save-dev
npm install @prisma/client@latest

**Comment by muhammad-fiaz** [comment]
Using a VPN temporarily fixed the Prisma engine download issue for me. It looks like the problem was network-related (possibly regional restrictions or unstable connection). After switching to VPN, npx prisma generate completed successfully.

**Comment by mohdsaadshaikh** [comment]
`generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "windows"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
`

This works for me
